equinix/fabric_connection_schema.go

```diff
@@ -0,0 +1,840 @@
+			Optional:    true,
+			Description: "Vlan Customer Tag information, vlanCTag value specified for QINQ connections",
+		},
+		"vni": {
+			Type:        schema.TypeInt,
+			Optional:    true,
+			Description: "vni",
+		},
+	}
+}
+
+var createAccessPointVirtualDeviceRes = &schema.Resource{
+	Schema: createAccessPointVirtualDeviceSch(),
+}
+
+func createAccessPointVirtualDeviceSch() map[string]*schema.Schema {
+
+	return map[string]*schema.Schema{
+		"href": {
+			Type:        schema.TypeString,
+			Computed:    true,
+			Description: "Unique Resource Identifier",
+		},
+		"uuid": {
+			Type:        schema.TypeString,
+			Optional:    true,
+			Description: "Equinix-assigned Virtual Device identifier",
+		},
+		"type": {
+			Type:        schema.TypeString,
+			Computed:    true,
+			Description: "Virtual Device type",
+		},
+	}
+}
+
+func createAccessPointInterface() map[string]*schema.Schema {
+	return map[string]*schema.Schema{
+		"uuid": {
+			Type:        schema.TypeString,
+			Optional:    true,
+			Description: "Equinix-assigned interface identifier",
+		},
+		"id": {
+			Type:        schema.TypeString,
+			Computed:    true,
+			Description: "id",
+		},
+		"type": {
+			Type:        schema.TypeString,
+			Computed:    true,
+			Description: "Interface type",
+		},
+	}
+}
+
+func createFabricConnectionRoutingProtocol() map[string]*schema.Schema {
+	return map[string]*schema.Schema{
+		"type": {
+			Type:        schema.TypeString,
+			Optional:    true,
+			Description: "Routing Protocol type",
+		},
+		"uuid": {
+			Type:        schema.TypeString,
+			Optional:    true,
+			Description: "Routing protocol instance identifier",
+		},
+		"state": {
+			Type:        schema.TypeString,
+			Optional:    true,
+			Description: "Routing protocol instance state",
+		},
+	}
+}
+
+var createPortRes = &schema.Resource{
+	Schema: createPortSch(),
+}
+
+func createPortSch() map[string]*schema.Schema {
+	return map[string]*schema.Schema{
+		"uuid": {
+			Type:        schema.TypeString,
+			Optional:    true,
+			Description: "Port information",
+		},
+		"href": {
+			Type:        schema.TypeString,
+			Computed:    true,
+			Description: "Unique Resource Identifier",
+		},
+		"name": {
+			Type:        schema.TypeString,
+			Optional:    true,
+			Description: "Port name",
+		},
+		"redundancy": {
+			Type:        schema.TypeSet,
+			Optional:    true,
+			Description: "Redundancy Information",
+			Elem: &schema.Resource{
+				Schema: createPortRedundancySch(),
+			},
+		},
+	}
+}
+
+var createConnectionSideAccessPointRes = &schema.Resource{
+	Schema: createConnectionSideAccessPointSch(),
+}
+
+func createConnectionSideAccessPointSch() map[string]*schema.Schema {
+	return map[string]*schema.Schema{
+		"type": {
+			Type:         schema.TypeString,
+			Optional:     true,
+			ValidateFunc: validation.StringInSlice([]string{"COLO", "VD", "VG", "SP", "IGW", "SUBNET", "GW"}, true),
+			Description:  "Access point type",
+		},
+		"authentication_key": {
```

Comment:
Is this is sensitive, add `Sensitive: true,`

Reply:
This will prevent the field from being exposed in Terraform output among other things.
https://www.terraform.io/plugin/sdkv2/best-practices/sensitive-state

Reply:
These are defined as Enums however, I see API accepting lower cases values. Hence, marked "ignorecase" as true.